server/server.go

```diff
@@ -2632,14 +2587,29 @@ func (s *StanServer) setSubStartSequence(cs *stores.ChannelStore, sub *subState,
 			sub.ClientID, sub.subject)
 	case pb.StartPosition_TimeDeltaStart:
 		startTime := time.Now().UnixNano() - sr.StartTimeDelta
+		// If there is no message, seq will be 0.
 		seq := s.getSequenceFromStartTime(cs, startTime)
 		if seq > 0 {
+			// If the time delta is in the future relative to the last
+			// message in the log, 'seq' will be equal to last sequence + 1,
+			// so this would translate to "new only" semantic.
 			lastSent = seq - 1
 		}
 		Debugf("STAN: [Client:%s] Sending from time, subject=%s time=%d seq=%d",
 			sub.ClientID, sub.subject, startTime, lastSent)
 	case pb.StartPosition_SequenceStart:
-		if sr.StartSequence > 0 {
+		// If there is no message, firstSeq and lastSeq will be equal to 0.
+		firstSeq, lastSeq := cs.Msgs.FirstAndLastSequence()
+		// StarSequence is an uint64, so can't be lower than 0.
+		if sr.StartSequence < firstSeq {
```

Comment:
I'd suggest a debug message.  Perhaps even an option to dictate behavior here?  The concern lies in the case where a log rollover occurred - some users would want to know they are not receiving all messages in their stream.